lib/Service/StackService.php

```diff
@@ -362,4 +379,110 @@ public function reorder($id, $order) {
 
 		return $result;
 	}
+
+	/**
+	 * @param $id
+	 * @param $boardId
+	 * @param $userId
+	 * @return Stack
+	 * @throws StatusException
+	 * @throws BadRequestException
+	 */
+	public function clone($id, $boardId, $userId) {
```

Comment:
Actually let's remove the userId as a parameter and just inject the current one through the constructor

Reply:
How to do this?

Reply:
https://docs.nextcloud.com/server/latest/developer_manual/basics/dependency_injection.html for general details about dependency injection. You'll need to add userId to the constructor